Dev

vanhauser-thc opened the pull request.

retrage committed:
Add initial CoreSight mode support

The original code is:
https://github.com/RICSecLab/AFLplusplus-cs/tree/retrage/coresight-mode-pr

Signed-off-by: Akira Moroo <retrage01@gmail.com>

domenukk commented on src/afl-common.c:
```diff
@@ -51,6 +55,66 @@ u8  last_intr = 0;
   #define AFL_PATH "/usr/local/lib/afl/"
 #endif
 
+u32 check_binary_signatures(u8 *fn) {
```
I (still) don't understand why we need this, the compiled target knows if it is using a deferred forkserver or not...
There should be a better way than to have the parent scan the binary, imho...

vanhauser-thc replied:
yeah maybe. we can put that on the todo list and do something about this at some point. eg. add a change to a global var in afl-compiler-rt.o.c in the __AFL_INIT macro etc.

vanhauser-thc committed:
fix gcc warning

devnexen committed:
frida mode musl build fix

vanhauser-thc committed:
Merge pull request #1128 from devnexen/frida_mode_musl_build_fix

frida mode musl build fix

devnexen committed:
frida mode musl further changes display stack trace

vanhauser-thc committed:
Merge pull request #1129 from devnexen/frida_mode_musl_further

frida mode musl further changes display stack trace

wtdcode committed:
Update real_map_size when doing a realloc

vanhauser-thc committed:
Merge pull request #1130 from wtdcode/dev

Wrong map size when a realloc happens

vanhauser-thc committed:
variants

Commit:
Fix issues with Yama restrictions in FRIDA mode

vanhauser-thc committed:
fix

vanhauser-thc committed:
Merge pull request #1131 from WorksButNotTested/frida-debug

Fix issues with Yama restrictions in FRIDA mode

vanhauser-thc committed:
fix

devnexen committed:
frida mode macOs build fix proposal

vanhauser-thc committed:
Merge pull request #1132 from devnexen/frida_macos_build_fix

frida mode macOs build fix proposal

adrianherrera committed:
optimin: fix cmake flags

Disable RTTI. Fixes issue #1133

vanhauser-thc committed:
Merge pull request #1134 from adrianherrera/bugfix/optimin-cmake

optimin: fix cmake flags

vanhauser-thc committed:
gcc instrumentation opt

vanhauser-thc committed:
variant 6

kcwu committed:
fix off by one bug of PCGUARD instrumentation

fix bug #1120

vanhauser-thc committed:
Merge pull request #1135 from kcwu/fix-pcguard-off-by-1

fix off by one bug of PCGUARD instrumentation

llzmb committed:
Edit list of environment variables

llzmb committed:
Fix line breaks

vanhauser-thc committed:
help gcc

vanhauser-thc committed:
add cmplog compatability check

vanhauser-thc committed:
wipe overflow data

vanhauser-thc committed:
insert select instrumentation

vanhauser-thc committed:
fix

vanhauser-thc committed:
fix

vanhauser-thc committed:
remove debug

vanhauser-thc committed:
remove debug

vanhauser-thc committed:
support select with vectors

vanhauser-thc committed:
select support for LTO

vanhauser-thc committed:
nosan

vanhauser-thc committed:
fix

vanhauser-thc committed:
modify frida for new cmplog

vanhauser-thc committed:
support vectorized cmps

vanhauser-thc committed:
vectorized coverage only possible for llvm 14 :(

vanhauser-thc committed:
support llvm >= 11

vanhauser-thc committed:
fix

vanhauser-thc committed:
afl4 qemu

yuawn committed:
Fix request size & remove redundant code (#1139)

* fix request size

* fix null terminator index

* remove redundant code

llzmb committed:
Clean up references to power_schedules.md

vanhauser-thc committed:
Merge pull request #1140 from llzmb/docs_cleanup_references_power_schedules

Clean up references to power_schedules.md

vanhauser-thc committed:
move bitmap and saveauto

vanhauser-thc committed:
remove overflow wiping

vanhauser-thc committed:
fix unneeded ifdef

vanhauser-thc committed:
Merge pull request #1137 from AFLplusplus/select

select instrumentation for coverage and cmplog

vanhauser-thc committed:
Merge pull request #1142 from AFLplusplus/dev

Dev

vanhauser-thc committed:
remove old var

vanhauser-thc committed:
update todo

vanhauser-thc committed:
add AFL_USE_TSAN

vanhauser-thc committed:
update for new sanitizer support

devnexen committed:
afl-as: missing one format spec (for tsan perharps)

vanhauser-thc committed:
Merge pull request #1144 from devnexen/afl_as_fix

afl-as: missing one format spec (for tsan perharps)

yuawn committed:
unify output format

vanhauser-thc committed:
Merge pull request #1145 from yuawn/dev

Unify output format

vanhauser-thc committed:
http->https

vanhauser-thc committed:
update qemafl

vanhauser-thc committed:
better error report for old qemu/frida cmplog

vanhauser-thc committed:
better screen update

vanhauser-thc committed:
update changelog

vanhauser-thc committed:
30% faster hashing speed

llzmb committed:
Edit list of environment variables

tl455047 committed:
Fix index of cmp_map->log[key].

tl455047 committed:
code-format

vanhauser-thc committed:
Merge pull request #1148 from tl455047/dev

Fix index of cmp_map->log[key].

vanhauser-thc committed:
Merge pull request #1141 from AFLplusplus/afl4

cmplog enhancement variant

vanhauser-thc committed:
env

DMaroo committed:
Use FIFOs instead of hacky sleep code to pass messages

DMaroo committed:
Cleanup earlier

 * Exit if mkfifo fails
 * Use .tmp so that in any case the cleanup fails, user doesn't have to
   see this dir unless explicitly listing hidden files

vanhauser-thc committed:
Merge pull request #1149 from DMaroo/ui-fifo

Use FIFOs instead of hacky sleep code to pass messages in afl-plot

vanhauser-thc committed:
nits

Commit:
Suppress coverage for deterministic branches

Commit:
Add support for option to disable backpatch

Commit:
Changes to cmplog to use hashes block ID

Commit:
Added performance test

vanhauser-thc committed:
Merge pull request #1150 from WorksButNotTested/perf-final

Perf final

vanhauser-thc committed:
update qemuafl

vanhauser-thc committed:
fix afl- tools for new frida

Commit:
Suppress spurious output

Commit:
Inline instrumentation optimization for x64

llzmb committed:
Edit list of environment variables

adrianherrera committed:
optimin: fix optimin git clone

vanhauser-thc committed:
Merge pull request #1153 from WorksButNotTested/frida

Frida

vanhauser-thc committed:
Merge pull request #1154 from adrianherrera/optimin/build-script-fix

optimin: fix optimin git clone

vanhauser-thc committed:
doc

vanhauser-thc committed:
update qemuafl, update changelog

Commit:
Minor change to inline assembly

Commit:
Fix block ID handling

vanhauser-thc committed:
Merge pull request #1155 from WorksButNotTested/frida

Frida

Commit:
Changes to fix use of unstable coverage

Commit:
Fix to handling of persistent mode using JS

vanhauser-thc committed:
Merge pull request #1157 from WorksButNotTested/frida

Frida

retrage committed:
Clarify usage message for ARM CoreSight mode

REF:
https://github.com/AFLplusplus/AFLplusplus/pull/1156#issuecomment-966196217

Signed-off-by: Akira Moroo <retrage01@gmail.com>

retrage committed:
Remove unnecessary mem_limit in afl-showmap

Signed-off-by: Akira Moroo <retrage01@gmail.com>

retrage committed:
Add platform check for `-A` CoreSight mode

REF:
https://github.com/AFLplusplus/AFLplusplus/pull/1156#discussion_r747454306

Signed-off-by: Akira Moroo <retrage01@gmail.com>

retrage committed:
Fix platform check for `-A` CoreSight mode

Signed-off-by: Akira Moroo <retrage01@gmail.com>

retrage committed:
Check `-M` / `-S` is not specified with `-A`

Signed-off-by: Akira Moroo <retrage01@gmail.com>

vanhauser-thc committed:
cmplog fix

invalid-email-address committed:
Changes to allow configuration of stalker adjacent blocks

invalid-email-address committed:
Further optimization to coverage code

vanhauser-thc committed:
Merge pull request #1162 from WorksButNotTested/frida

Frida

vanhauser-thc committed:
Merge pull request #1136 from llzmb/docs_edit_environment_variables

Edit list of environment variables

retrage committed:
Fix finding glibc patches

Signed-off-by: Akira Moroo <retrage01@gmail.com>

retrage committed:
coresight_mode: Add TODO list to README.md

Signed-off-by: Akira Moroo <retrage01@gmail.com>

vanhauser-thc committed:
Merge pull request #1156 from retrage/retrage/coresight-mode-pr

Add CoreSight mode support

vanhauser-thc committed:
showmap -A -> -H, accurate help output

vanhauser-thc committed:
fix afl-showmap

vanhauser-thc committed:
add coresight to docs

vanhauser-thc committed:
nit

vanhauser-thc committed:
d2

invalid-email-address committed:
Disable fetching adjacent blocks when coverage is enabled

invalid-email-address committed:
Fixes to exclude ASAN DSO

invalid-email-address committed:
Fix sorting of ranges

invalid-email-address committed:
Add __afl_set_persistent_mode

invalid-email-address committed:
Bump FRIDA version

vanhauser-thc committed:
Merge pull request #1164 from WorksButNotTested/fix

Fix

invalid-email-address committed:
Added test of JS hooking LLVMFuzzerTestOneInput

invalid-email-address committed:
Improve JS bindings for hooking functions

invalid-email-address committed:
Fix ASAN DSO exclusion

vanhauser-thc committed:
ensure default sigchild handler in afl-compiler-rt

kcwu committed:
fork server: restore SIGCHLD handler for child process

vanhauser-thc committed:
Merge pull request #1168 from kcwu/restore-sigchld-handler

fork server: restore SIGCHLD handler for child process

vanhauser-thc committed:
Merge pull request #1167 from WorksButNotTested/js

Js

vanhauser-thc committed:
cmplog fix

invalid-email-address committed:
Optimize assembly for x86

invalid-email-address committed:
Fix freetype target to build for x86 (32-bit)

invalid-email-address committed:
Added vorbis test

invalid-email-address committed:
Compilation warning fixes

invalid-email-address committed:
Shift saved values down the stack by 8 bytes

invalid-email-address committed:
Minor ASAN fix (again)

invalid-email-address committed:
Changes to skip coverage code on deterministic branches on x86

vanhauser-thc committed:
Merge pull request #1171 from WorksButNotTested/frida

Frida

hexcoder- committed:
cmplog: Fix compilation for LLVM 12.0.0 (getFixedValue was not available here)

hexcoder- committed:
Merge branch 'dev' of https://github.com/AFLplusplus/AFLplusplus into dev

hexcoder- committed:
Fix compilation on RaspberryPi 32-bit (please review)

hexcoder- committed:
Fix compilation error under RaspberryPi 64-Bit Manjaro Linux (no
SYS_eventfd available here)

vanhauser-thc committed:
enhance cmplog rtn hooks

llzmb committed:
Clean up docs folder

retrage committed:
coresight_mode: Change to use https for submodule URLs

Signed-off-by: Akira Moroo <retrage01@gmail.com>

vanhauser-thc committed:
Merge pull request #1173 from retrage/retrage/coresight-mode-pr

coresight_mode: Change to use https for submodule URLs

llzmb committed:
Merge "binaryonly_fuzzing.md" into "fuzzing_binary-only_targets.md"

llzmb committed:
Clean up docs folder

llzmb committed:
Clean up docs folder

hexcoder- committed:
add supported boards, name prerequisite

vanhauser-thc committed:
better string length counting

vanhauser-thc committed:
better string length counting

hexcoder- committed:
Merge branch 'dev_newpm' into dev

llzmb committed:
Clean up docs folder

llzmb committed:
Delete obsolete files

llzmb committed:
Merge ctx and ngram into llvm, fix references

llzmb committed:
Edit "README.persistent_mode.md"

vanhauser-thc committed:
fix for compcov transform strings

llzmb committed:
Edit "README.persistent_mode.md"

vanhauser-thc committed:
llvm new passmanager fixes

vanhauser-thc committed:
further fixes for new pass manager

vanhauser-thc committed:
further fixes for new pass manager

vanhauser-thc committed:
do not add zero byte on string compares with len

vanhauser-thc committed:
fixes

llzmb committed:
Edit instrumentation READMEs

hexcoder- committed:
fix switch to new pass manager

vanhauser-thc committed:
fix for llvm < 11

llzmb committed:
Update docs/fuzzing_binary-only_targets.md

hexcoder- committed:
transform cmplog-instructions pass to new pass manager

llzmb committed:
Merge various files into "fuzzing_in_depth.md"

llzmb committed:
Merge branch 'docs_cleanup_folder_2' of https://github.com/llzmb/AFLplusplus into docs_cleanup_folder_2

llzmb committed:
Merge "common_sense_risks.md" into "fuzzing_in_depth.md"

llzmb committed:
Edit "fuzzing_in_depth.md"

vanhauser-thc committed:
update

llzmb committed:
Merge "perf_tips.md" into "best_practices.md" and "fuzzing_in_depth.md"

llzmb committed:
Edit instrumentation READMEs

vanhauser-thc committed:
update doc2

vanhauser-thc committed:
Merge pull request #1181 from AFLplusplus/d2

project report for GSoD

llzmb committed:
Edit FRIDA mode README.md

llzmb committed:
Edit QEMU mode READMEs

llzmb committed:
Change "AFL" to "AFL++" in "README.llvm.md", fix references

llzmb committed:
Fix spelling in FRIDA mode README.md

llzmb committed:
Edit FRIDA mode DEBUGGING.md

llzmb committed:
Fix typo (#1183)

llzmb committed:
Merge "ci_fuzzing.md" into "fuzzing_in_depth.md"

llzmb committed:
Edit FRIDA mode MD files

llzmb committed:
Fix typos

vanhauser-thc committed:
remove new llvm pass manager :(

invalid-email-address committed:
Added bloaty test

vanhauser-thc committed:
Merge pull request #1184 from WorksButNotTested/bloaty

Added bloaty test

Commit:
[fix] Custom mutator does not implement all three trim APIs, standard trimming will be used, but now the actual implementation does not match the description, fix this problem

eriktews committed:
Include cmake and meson in the Dockerfile

The README contains an examples how afl can be used with cmake, but
cmake is not included in the Dockerfile. Thus, when running afl using
docker, one cannot invoke cmake from within the container.

This commit adds cmake and meson to the list of packages in the
Dockerfile that are installed when the docker image is generated.

vanhauser-thc committed:
Merge pull request #1185 from eriktews/cmake-in-docker-container

Include cmake in the Dockerfile

vanhauser-thc committed:
Merge pull request #1186 from eternalsakura/stable

[fix] Fix custom mutator trim bug

vanhauser-thc committed:
Merge pull request #1177 from llzmb/docs_cleanup_instrumentation_folder

Clean up instrumentation folder

invalid-email-address committed:
Fix some minor formatting

invalid-email-address committed:
Update docs

invalid-email-address committed:
Fix transposed configuration options

invalid-email-address committed:
Add support for parsing zero as a configuration setting

invalid-email-address committed:
Changes to support defaults for configuration options

llzmb committed:
Remove step 5 (interpret output) of quick start

llzmb committed:
Merge branch 'dev' into docs_cleanup_folder_2

llzmb committed:
Merge branch 'dev' into docs_cleanup_folder_2

vanhauser-thc committed:
Merge pull request #1172 from llzmb/docs_cleanup_folder_2

Clean up docs folder

vanhauser-thc committed:
Merge pull request #1187 from WorksButNotTested/frida

Frida

llzmb committed:
Edit QEMU mode README.md

llzmb committed:
Merge branch 'dev' into docs_edit_readme_frida_mode_qemu_mode

vanhauser-thc committed:
Merge pull request #1182 from llzmb/docs_edit_readme_frida_mode_qemu_mode

Edit FRIDA mode and QEMU mode READMEs

vanhauser-thc committed:
update docs

vanhauser-thc committed:
unify LLVM_VERSION_... usage

vanhauser-thc committed:
remove quickly

domenukk committed:
New unicornafl rust bindings (#1188)

* uc bindings

* fixed new uc rust bindings

* updated to latest uc1

* updated rust bindings

domenukk commented on TODO.md:
```diff
@@ -1,30 +1,27 @@
 # TODO list for AFL++
 
-## Roadmap 3.00+
+## TODO
```
TODO is already in the title (minor nitpick)

domenukk commented on src/afl-fuzz-queue.c:
```diff
@@ -315,7 +315,96 @@ void mark_as_redundant(afl_state_t *afl, struct queue_entry *q, u8 state) {
+  while (offset < len) {
+
+    // ASCII: <= 0x7F to allow ASCII control characters
+    if ((buf[offset + 0] == 0x09 || buf[offset + 0] == 0x0A ||
```
why `+ 0` here?

vanhauser-thc replied:
just for copy-paste-ability + readability

vanhauser-thc committed:
todo

vanhauser-thc committed:
fixes

llzmb committed:
Remove the word "simply"

llzmb committed:
Remove the word "please"

llzmb committed:
Change the word "chapter" to "section"

llzmb committed:
Change the word "behaviour" (BE) to "behavior" (AE)

llzmb committed:
Change the word "entrypoint" to "entry point"

llzmb committed:
Change the word "env var" to "environment variable"

llzmb committed:
Change "AFLplusplus" to "AFL++"

invalid-email-address committed:
Fixes for arm32

vanhauser-thc committed:
Merge pull request #1189 from WorksButNotTested/arm32

Fixes for arm32

llzmb committed:
Change "AFL" to "AFL++"

llzmb committed:
Fix spelling of words related to "build"

llzmb committed:
Remove the word "we"

llzmb committed:
Fix punctuation in connection with "however"

llzmb committed:
Fix punctuation in connection with "e.g."

llzmb committed:
Change "eg" to "e.g." and fix punctuation

llzmb committed:
Fix typos - 1st run

vanhauser-thc committed:
what to do with dlopen libs?

vanhauser-thc committed:
change dlopen solution

Kiprey committed:
Fix memory leak in libprotobuf-mutator-example

hexcoder- committed:
Fix compilation for LLVM 12.0.0, use older method getKnownMinValue()

llzmb committed:
Remove old references

llzmb committed:
Fix punctuation in connection with "etc."

llzmb committed:
Fix line length and formatting

llzmb committed:
Fix punctuation in connection with "by default"

llzmb committed:
Fix punctuation in connection with "with"

llzmb committed:
Fix typos - 2nd run

llzmb committed:
Fix spelling of "FRIDA mode" and "QEMU mode"

llzmb committed:
Add links to orphaned files

llzmb committed:
Add missing blank lines and remove double blank lines

devnexen committed:
libtokencap: warning build fix.

vanhauser-thc committed:
Merge pull request #1193 from devnexen/libtoken_warn_fix

libtokencap: warning build fix.

vanhauser-thc committed:
Merge pull request #1192 from Kiprey/stable

Fix memory leak in libprotobuf-mutator-example

llzmb committed:
Fix broken links - 1st run

llzmb committed:
Fix various missed issues - 1st run

llzmb committed:
Fix usage and punctuation in connection with "then" and "than"

WorksButNotTested committed:
Bump FRIDA version (#1197)

Co-authored-by: Your Name <you@example.com>

vanhauser-thc committed:
Merge pull request #1191 from llzmb/docs_quality_assurance

Docs content - quality assurance

vanhauser-thc committed:
fix AFL_REAL_LD for afl-cc

Microsvuln committed:
Update build-codeql.sh

Update codeql version to 2.7.3

vanhauser-thc committed:
Merge pull request #1199 from Microsvuln/patch-1

Update CodeQL to latest
